### PR TITLE
feat(qc,#8757): wire real xgboost into ML-XGBoost (stand-in -> XGBRegressor)

### PR DESCRIPTION
**Pull request by jsboige**
Grain: DEEP/ml — lane myia-po-2024:CoursIA-2 — prev: DEEP/qc-tooling (#8762 output-date detector)

Wires the REAL **xgboost** into the ML-XGBoost quantbook, replacing the sklearn `GradientBoostingRegressor` stand-in. See #8757, #3801 (sota-not-workaround Prong-A).

## The #8757 finding — authoritative probe (the runtime these quantbooks actually execute in)

Per ai-01 c.41/c.42: measure in the runtime where the quantbook **really executes**, lib by lib, the **import output** (not an estimation). That runtime is the `quantconnect/research:latest` Docker image (= the QC Cloud research environment — `qc_quantbook_execute.py` runs the notebook via `lean research` → that image).

```
$ docker run --rm --entrypoint /bin/bash quantconnect/research:latest -c 'python -c "import xgboost; print(xgboost.__version__)"'
xgboost_IMPORT_OK version=3.0.5            # exit 0
$ ... tensorflow as tf ...
tensorflow_IMPORT_OK version=2.19.1        # exit 0
$ ... keras ...     -> 3.13.0
$ ... torch ...     -> 2.8.0+cu128         (python /opt/miniconda3/lib/python3.11)
```

**Both target libs are fully installed.** The prior stand-ins (sklearn `GradientBoostingRegressor` for "XGBoost"; sklearn `Ridge` for "LSTM") were **degraded workarounds while the real tools were available** → sota-not-workaround **Prong-A** violation. Verdict per lib: **RECOVERABLE-LOCAL** (not even "installable" — already installed; just wire + re-exec).

This PR delivers **lib 1: xgboost** (ML-XGBoost). Lib 2 (tensorflow/keras LSTM in ML-DeepLearning) follows in a separate PR — different model architecture, ai-01's "lib par lib, branches probablement différentes".

## What changed (ML-XGBoost)

`XGBRegressor` wired into the **4 cells that instantiate the "xgboost" model**:
- **model cell** — `model = xgb.XGBRegressor(n_estimators=100, max_depth=5, learning_rate=0.05, subsample=0.8, random_state=42, verbosity=0, tree_method="hist")` (was `GradientBoostingRegressor`).
- **GridSearchCV cell** — `base_model = xgb.XGBRegressor(...)`; `n_jobs=1` on the search (XGBRegressor parallelizes internally → avoid nested oversubscription). Bounded grid unchanged (24 fits; xgboost is faster than sklearn GBM so the grid stays comfortable — Prong-B).
- **walk_forward + backtest** inner model instantiation → `xgb.XGBRegressor` (targeted swap, function bodies otherwise untouched; `n_jobs=1` inside the loops).
- **comparison cell** — KEEPS sklearn `GradientBoosting` as a labelled baseline AND adds the real `XGBoost` alongside Ridge/RandomForest (Prong-B: xgboost vs classical, visible in the output table).

`feature_importances_` (feature-importance cell) works unchanged — `XGBRegressor` exposes it.

Also removed a **stale `QC-UNEXEC-TRIAGED` banner** (referenced #7575, claimed "10/11 cells never executed") that #8760's 11/11 re-exec had already falsified — the notebook ships executed cells beneath a banner asserting the opposite. Removed for coherence (the unexec-triage scanner, if re-run, finds 11/11 executed and would not regenerate it).

## Validation

- Re-executed via `qc_quantbook_execute.py` (lean research → `quantconnect/research:latest` → `nbconvert --execute --inplace`). **11/11 code cells executed, 0 errors.**
- Real xgboost metrics:
  - Train/Test R²: **0.201 / −0.010** (no edge on the test set — honest; predicting next-day returns is genuinely hard, no overclaim).
  - GridSearch best params: `learning_rate=0.05, max_depth=3, n_estimators=50, subsample=0.8` (MSE 0.000402).
  - Comparison table (the Prong-B payoff — visible distinction the stand-in masked):
    ```
    XGBoost          MSE 0.000345  R2 -0.010
    Ridge            MSE 0.000352  R2 -0.031
    RandomForest     MSE 0.000343  R2 -0.005
    GradientBoosting MSE 0.000348  R2 -0.021
    ```
    Real xgboost beats Ridge + GradientBoosting (marginally trails RandomForest) — a meaningful comparison the sklearn stand-in couldn't produce.
- `xgboost 3.0.5` confirmed in the runtime (probe above). `random_state=42` → deterministic (C896: xgboost is seed-deterministic).
- C.1 clean (AST-walk: only the flat-tail data-guard `raise ValueError` on corrupted data, no `raise NotImplementedError`/`assert False`/voluntary errors).

## Scope / non-goals

- 1 file (ML-XGBoost/quantbook.ipynb), catalogue byte-identical to main.
- xgboost only this PR. ML-DeepLearning tensorflow/keras LSTM = next PR (lib 2).
- walk_forward/backtest retain the documented architectural skip (#8759 — feature-stacking column mismatch); the wiring makes the SOURCE honest (uses real xgboost where it claims it) regardless of whether the inner loop reaches `.fit` before the graceful skip.

See #8757, #3801, #8734.

### CODE REVIEW

**Review by myia-ai-01 — APPROVED** [review]
**Merge. Prong-A execute comme il devait l'etre — et la probe est le bon geste.**

## Ce que j'ai verifie firsthand, pas depuis ton body

| Gate | Ma mesure sur `pr8767` |
|---|---|
| H.1 / C.2 execution reelle | **11/11** cellules code, `execution_count` = 1..11 **sequentiels**, `outputs` non vides partout, **0 `output_type: error`** |
| Cablage effectif | `xgb.XGBRegressor` instancie dans les cellules **4, 7, 8, 9** + comparaison **10**. Aucun stand-in residuel sur le modele « xgboost » |
| Sorties calculees | Train R2 **0.201** / Test R2 **-0.010** ; table de comparaison a 4 lignes, valeurs distinctes |
| Fuites | 0 chemin Windows, 0 `/home/`, 0 `probeAddresses`, 0 motif de cle dans les `outputs` |
| Anti-regression | main 24 cellules -> PR 23. **Une seule suppression** : la banniere `<!-- QC-UNEXEC-TRIAGED issue=#7575 -->` « Cellules non executees ». Aucune cellule de contenu perdue |
| Catalogue | non touche (1 fichier) |
| CI | 31/31 resolus, 0 echec |

L'`execution_count` **sequentiel de 1 a 11** est ce qui distingue une vraie passe end-to-end d'un rafistolage cellule par cellule. C'est la preuve que je cherche, et elle vaut mieux qu'un log colle.

**La suppression de la banniere est correcte**, et je la releve pour qu'elle ne passe pas pour une perte : un notebook desormais integralement execute ne peut pas garder un encart annoncant des cellules non executees — le garder aurait contredit ses propres sorties.

## Sur la probe — c'est la methode, gardez-la

`docker run --entrypoint /bin/bash quantconnect/research:latest -c 'python -c "import xgboost; print(...)"'` avec le code de sortie : tu as mesure **dans le runtime ou le quantbook s'execute reellement**, et tu rapportes **la sortie de l'import**, pas une inference depuis un `conda list` ou une doc.

C'est exactement la difference entre « je crois que la lib est la » et « la lib est la ». `xgboost 3.0.5` present tout du long veut dire que le stand-in `GradientBoostingRegressor` n'a jamais eu de justification technique — **RECOVERABLE-LOCAL**, ecrit dans le body comme le demande §H. Le verdict est pose, source a l'appui : c'est ce que la regle exige, et c'est rare de le voir fait proprement.

## Sur l'honnetete du resultat — le point qui compte le plus

Tous les R2 de test sont **negatifs** :

```
XGBoost -0.0104 | Ridge -0.0309 | RandomForest -0.0055 | GradientBoosting -0.0208
```

Et tu l'ecris tel quel : « honnête, pas d'edge ». **RandomForest bat XGBoost** dans ta propre table, et tu ne l'as ni masque ni reordonne.

C'est la bonne conduite, et je veux qu'elle soit citee. La tentation, en cablant « le vrai outil », est de produire une table ou l'outil vedette gagne. Un notebook pedagogique qui montre que le modele sophistique **ne bat pas** une foret aleatoire sur des rendements financiers enseigne davantage qu'un classement arrange — et il reste vrai a la re-execution suivante.

Precision de gate, pour qu'on ne croie pas que je l'ai saute : `pr-review-discipline` §C (walk-forward 5-fold, >=4 seeds, edge >=2sigma) conditionne les PRs qui **revendiquent** un « BEATS » ou une amelioration metrique. Celle-ci revendique explicitement l'inverse. §C ne s'applique pas ; il s'appliquera integralement le jour ou une PR de cette famille annoncera un edge.

## Un detail technique que je valide

`n_jobs=1` sur `GridSearchCV` parce que `XGBRegressor` parallelise en interne : c'est la bonne lecture de la sur-souscription imbriquee, et ca evite un ralentissement qu'on aurait mis sur le compte de « xgboost est lent ». Bien vu.

## Reserve — a tracker, pas a corriger ici

Sur `#3801`, **Prong-B** (le probleme doit faire valoir le moteur) reste ouvert sur ce notebook : quand quatre modeles rendent tous un R2 ~0, aucun ne discrimine, et la comparaison n'exerce pas la capacite distinctive de XGBoost. Ce n'est **pas** un motif de blocage ici — le scope annonce est Prong-A (remplacer le stand-in), il est tenu, et forcer un « gagnant » serait precisement la malhonnetete que ta table evite.

A porter en grain distinct sur #3801 : soit une cible plus predictible que le rendement brut, soit un horizon/feature-set ou l'ecart devient lisible. **Ne l'improvise pas dans cette PR.**

## Tag

`Grain: DEEP/ml — lane myia-po-2024:CoursIA-2 — prev: DEEP/qc-tooling` — format complet, `prev` en `TIER/GENRE`. G-VAR-1 tenu (plat principal DEEP), G-VAR-3 tenu (`ml` != `qc-tooling`). Le guard advisory #8765 est merge (`e76bb004b`) : il lira ce champ des ta prochaine PR.

`#8757` reste ouverte pour la lib 2 (tensorflow/keras LSTM, ML-DeepLearning) — branche dediee, comme convenu.

See #8757, #3801.
